File: hackerrank/Software Engineer Prep Kit/swift/count_elements_greater_than_previous_average.swift
import Foundation



/*
 * Complete the 'countResponseTimeRegressions' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY responseTimes as parameter.
 */

func countResponseTimeRegressions(responseTimes: [Int]) -> Int {
    
    var total = 0
    var average = 0
    var counter = 0

    for i in 1..<responseTimes.count  {

        total += responseTimes[i-1]
        average = total / i

        if (responseTimes[i] > average) {
            counter += 1

        }
    }
    
    return counter
}

guard let responseTimesCount = Int((readLine()?.trimmingCharacters(in: .whitespacesAndNewlines))!)
else { fatalError("Bad input") }

var responseTimes = [Int]()

for _ in 1...responseTimesCount {
    guard let responseTimesItem = Int((readLine()?.trimmingCharacters(in: .whitespacesAndNewlines))!)
    else { fatalError("Bad input") }

    responseTimes.append(responseTimesItem)
}

guard responseTimes.count == responseTimesCount else { fatalError("Bad input") }

let result = countResponseTimeRegressions(responseTimes: responseTimes)

print(result)

File: hackerrank/Software Engineer Prep Kit/python/count_elements_greater_than_previous_average.py
#!/bin/python3

import math
import os
import random
import re
import sys



#
# Complete the 'countResponseTimeRegressions' function below.
#
# The function is expected to return an INTEGER.
# The function accepts INTEGER_ARRAY responseTimes as parameter.
#

def countResponseTimeRegressions(responseTimes):
    total = 0
    average = 0
    counter = 0

    for i in range(1, len(responseTimes)):

        total += responseTimes[i-1]
        average = total / i

        if (responseTimes[i] > average):
            counter += 1
    
    
    return counter
    

if __name__ == '__main__':
    responseTimes_count = int(input().strip())

    responseTimes = []

    for _ in range(responseTimes_count):
        responseTimes_item = int(input().strip())
        responseTimes.append(responseTimes_item)

    result = countResponseTimeRegressions(responseTimes)

    print(result)
